backends/ebpf/ebpfControl.cpp

```diff
@@ -47,21 +47,65 @@ bool ControlBodyTranslator::preorder(const IR::PathExpression* expression) {
 }
 
 void ControlBodyTranslator::processFunction(const P4::ExternFunction* function) {
-    ::error("%1%: Not supported", function->method);
+    if (function->method->name.name == "assert") {
+        BUG_CHECK(function->expr->arguments->size() == 1,
+                "%1%: Expected one argument in ", function);
+        builder->emitIndent();
+        builder->append("if (");
+        visit(function->expr->arguments->at(0)->expression);
+        builder->append(" == false) ");
+        builder->blockStart();
+        builder->emitIndent();
+        builder->appendFormat("% s = % s;",
+             control->program->errorVar.c_str(), P4::P4CoreLibrary::instance.assertError);
+        builder->newline();
+        auto srcInfo = function->expr->srcInfo;
+        auto sourceFragment = srcInfo.toBriefSourceFragment();
+        auto sourceFile = srcInfo.getSourceFile();
+        auto linePosition = srcInfo.toPositionString();
+        builder->emitIndent();
+        builder->appendFormat(
+            "fprintf(stderr, \"Assertion error: ' % s ' failed, file ' % s ', line % s \\n\");",
```

Comment:
Have you tested this with kernel code? Can you just call fprintf from an ebpf program running in the kernel?

Reply:
I was trying to test it, but I couldn't manage to do it because I got some errors when I tried to compile .c file that is p4c output. So I did it according to the rest of code gen for ebpf backend.
Maybe I didn't understand well how to test it, so I would be thankful if you can give me more information  how to do it.

Reply:
If you get errors compiling a C file it's a bug in the ebpf back-end - please file an issue.
The ebpf testing tries to test the files in multiple ways, depending on your kernel version.
But the ebpf code is supposed to run in the kernel, so it may not use fprintf.
The more important question is how to handle an assertion failure in ebpf.